### ISSUE TITLE
Specific routes not working anymore when using next to wildcards

### ISSUE THREAD

**Issue by FireFrei** [pistacheio/pistache#278]
Hello!
I am using specific routes next to wildcard routes. this worked until commit 763d588ebaa579c1faefaedc4054f3a2281c081e (5761ab06383d7093b995e90ab9a895a394e78d24 is still working).

See my routes setup below:
```
/* Specific route */
Pistache::Rest::Routes::Get(mRouter, "/hello", Pistache::Rest::Routes::bind(&ApiRest::routeGetHelloWorld, this));

/* Static files (also index.html as /) via wildcard routes */
Pistache::Rest::Routes::Get(mRouter, "/", Pistache::Rest::Routes::bind(&ApiRest::routeGetStaticFile, this));
Pistache::Rest::Routes::Get(mRouter, "/*", Pistache::Rest::Routes::bind(&ApiRest::routeGetStaticFile, this));
Pistache::Rest::Routes::Get(mRouter, "/*/*", Pistache::Rest::Routes::bind(&ApiRest::routeGetStaticFile, this));
Pistache::Rest::Routes::Get(mRouter, "/*/*/*", Pistache::Rest::Routes::bind(&ApiRest::routeGetStaticFile, this));
```
When calling "/hello" now (since the mentioned commit above), the router uses the "/*" wildcard route to routeGetStaticFile() instead of routeGetHelloWorld().
The commit massage tells:
```
* Manually merged with master
* Fixed bugs
* invokeHandler modified in findRoute
* router_tests
```

By the way: Is there any better way of configuring a wildcard route with unspecified depth? At the moment (as you can see) I'm providing extra routes for every depth.

Cheers,
Matthias

**Comment by Fylax** [comment]
Hi, the problematic commit was part of a large PR. It was rolled back as there were various "hidden" bugs, like this.
Could you provide me informations about the compiler you used?
I should have fixed the bug, but I'd like to further investigate.

**Comment by FireFrei** [comment]
Sure!
`gcc version 7.3.0 (Ubuntu 7.3.0-16ubuntu3)`

**Comment by Fylax** [comment]
Hi!
I should have addressed the issue in #306

**Comment by Fylax** [comment]
PR merged. Could you try if it works now?
